### ISSUE TITLE
[Feature Request] Padding utility for field map in case tracking volume is larger than a given field map 

### ISSUE THREAD

**Issue by janheinri** [acts-project/acts#2523]
Hello,

While trying to use Acts for my setup, I noticed the following:
I am reading in a field map via the `acts.examples.MagneticFieldMapXyz` python binding. However I found out, that this throws an out of bounds error in line 206 in `Core/include/Acts/MagneticField/InterpolatedBFieldMap.hpp` due to the tracking volume being larger than the field map (I verified that these just occur on the edges or outside of the field map data):
`11:40:08    Propagator     ERROR     Step failed with MagneticFieldError:1: Interpolation out of bounds was requested`
I think it would be a nice addition to give the user the opportunity to pad the surrounding area of the field map with zero vectors in case the tracking volume is larger than the given field map data.
I temporarily solved it by returning a field cell filled with a zero vector in this case, but I am certain there is a much nicer solution to potentially implement this into the framework.

**Comment by paulgessinger** [comment]
Right, that's a very good observation. We actually had some discussion a while ago on whether interpolation should automatically, or if explicitly configured to, to fall back to a default value when the interpolation domain is exited.

I think we concluded not to have the interpolation do it silently, which I believe is also the right approach.

Having the ability to pad a grid seems useful to me. I imagine this would best be done when the grid is read, and require explicitly setting an increased interpolation domain volume.
